core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -353,6 +356,7 @@ protected TableMetadata refresh() {
   }
 
   @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
It didn't seem worth it to have the exception helper function just to avoid cyclomatic complexity. There's probably better refactoring we could do if we wanted to remove this, for now just decided to suppress warnings.